core/vm/tosca_ct_integration.go

```diff
@@ -0,0 +1,137 @@
+package vm
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/holiman/uint256"
+)
+
+// Error
+var ErrStopToken = errStopToken
+
+// Gas table
+func MemoryGasCost(mem *Memory, wordSize uint64) (uint64, error) {
+	return memoryGasCost(mem, wordSize)
+}
+
+// Stack
+func NewStack() *Stack {
+	return newstack()
+}
+
+func (st *Stack) Len() int {
+	return st.len()
+}
+
+func (st *Stack) Push(d *uint256.Int) {
+	st.push(d)
+}
+
+// EVM
+func (evm *EVM) GetDepth() int {
+	return evm.depth
+}
+
+func (evm *EVM) SetDepth(depth int) {
+	evm.depth = depth
+}
+
+// CallContext Call interceptor
+// CallContextInterceptor provides a basic interface for the EVM calling conventions. The EVM
+// depends on this context being implemented for doing subcalls and initialising new EVM contracts.
+// Based on ethereum's CallContext Interface
+type CallContextInterceptor interface {
+	// Call calls another contract.
+	Call(env *EVM, me ContractRef, addr common.Address, data []byte, gas uint64, value *uint256.Int) ([]byte, uint64, error)
+	// CallCode takes another contracts code and execute within our own context
+	CallCode(env *EVM, me ContractRef, addr common.Address, data []byte, gas uint64, value *uint256.Int) ([]byte, uint64, error)
+	// DelegateCall is same as CallCode except sender and value is propagated from parent to child scope
+	DelegateCall(env *EVM, me ContractRef, addr common.Address, data []byte, gas uint64) ([]byte, uint64, error)
+	// Create creates a new contract
+	Create(env *EVM, me ContractRef, data []byte, gas uint64, value *uint256.Int) ([]byte, common.Address, uint64, error)
+	// Create2 creates a new contract with a deterministic address
+	Create2(env *EVM, me ContractRef, code []byte, gas uint64, value *uint256.Int, salt *uint256.Int) ([]byte, common.Address, uint64, error)
+	// StaticCall executes a contract in read-only mode
+	StaticCall(env *EVM, me ContractRef, addr common.Address, input []byte, gas uint64) ([]byte, uint64, error)
+}
+
+// -- Interpreter Implementation Registry --
+
+// GethEVMInterpreter defines an interface for different interpreter implementations.
+type GethEVMInterpreter interface {
+	// Run the contract's code with the given input data and returns the return byte-slice
+	// and an error if one occurred.
+	Run(contract *Contract, input []byte, readOnly bool) (ret []byte, err error)
+}
+
+type InterpreterFactory func(evm *EVM, cfg Config) GethEVMInterpreter
+
+var interpreter_registry = map[string]InterpreterFactory{}
+
+func RegisterInterpreterFactory(name string, factory InterpreterFactory) {
+	interpreter_registry[strings.ToLower(name)] = factory
+}
+
+func NewInterpreter(name string, evm *EVM, cfg Config) GethEVMInterpreter {
```

Comment:
With this port, this function is introduced but not actually used.

Its role is to allow the Geth EVM to switch between its own interpreter implementation or any of the registered variants, including those offered by Tosca.

The integration happens here: https://github.com/Fantom-foundation/go-ethereum-substate/blob/b058188a043a40ac915d23549be6473aba7f0891/core/vm/evm.go#L147

Since this PR is introducing the registry infrastructure, it should also include its integration. It should be this one line linked above + an extra field in the EVM config.